### ISSUE TITLE
Dropdown example fails accessibility linting

### ISSUE THREAD

**Issue by kretzm** [telus/tds-core#422]
## Description
- The example implementation of the dropdown on the TDS side fails linting (jsx-a11y/label-has-for). It's not just the example, but any implementation that we have that uses the same HTML structure also throws the same error. I've set it to a warning in our app for now.
```
<div className="field field--quantity">
  <label htmlFor={`quantity-${product.offerId}`}>...</label>
  <div className="field__control field__control--dropdown">
    <select id={`quantity-${product.offerId}`} defaultValue={product.qty} onChange={...}>
      ...
    </select>
  </div>
</div>
```

## Reproduction Steps
View: https://tds.telus.com/#forms

## Version Number of TDS
- 0.23.0

*note attach screenshots whenever possible*
![screen shot 2017-10-06 at 9 49 07 am](https://user-images.githubusercontent.com/900447/31280927-f4488dda-aa7b-11e7-96a6-d2b30e03a189.png)
